src/components/QrcodeCapture/QrcodeCapture.vue

```diff
@@ -0,0 +1,153 @@
+<template>
+  <div :class="className">
+    <div :class="`${className}__activator`" @click="fileInput?.click()">
+      <slot />
+    </div>
+
+    <input
+      id="fileInput"
+      ref="fileInput"
+      type="file"
+      accept="image/*"
+      :class="`${className}__file-input`"
+      @change="onFileSelect"
+    />
+
+    <canvas ref="canvasElement" :class="`${className}__image`" />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { vibrate } from '@/lib/vibrate'
+import { useTemplateRef } from 'vue'
+import { IMG_MAX_SIZE } from '@/components/QrcodeCapture/consts'
+import type { BrowserQRCodeReader } from '@zxing/browser/esm/readers/BrowserQRCodeReader'
+
+const emit = defineEmits<{
+  (e: 'detect', text: string): void
+  (e: 'error', err: unknown): void
+}>()
+
+const className = 'qrcode-capture'
+
+let qrCodeText = ''
+let codeReader: BrowserQRCodeReader
+
+const canvas = useTemplateRef('canvasElement')
+const fileInput = useTemplateRef('fileInput')
+
+const drawCanvas = (file: File) => {
+  return new Promise((resolve, reject) => {
+    const img = new Image()
+    const imgUrl = URL.createObjectURL(file)
+
+    const revokeObjectURL = () => {
+      URL.revokeObjectURL(imgUrl)
+    }
+
+    img.onload = () => {
+      if (!canvas.value) {
+        revokeObjectURL()
+        return reject()
+      }
+
+      const ctx = canvas.value.getContext('2d')
+
+      if (!ctx) {
+        revokeObjectURL()
+        return reject()
+      }
+
+      const ratio = img.width / img.height
+      let newWidth = IMG_MAX_SIZE
+      let newHeight = IMG_MAX_SIZE / ratio
+
+      if (newHeight > IMG_MAX_SIZE) {
+        newHeight = IMG_MAX_SIZE
+        newWidth = IMG_MAX_SIZE * ratio
+      }
+
+      canvas.value.width = newWidth
+      canvas.value.height = newHeight
+
+      ctx.drawImage(img, 0, 0, canvas.value.width, canvas.value.height)
+
+      revokeObjectURL()
+
+      resolve(canvas)
+    }
+
+    img.onerror = (e) => reject(e)
+
+    img.src = imgUrl
+  })
+}
+
+/**
+ * Decode QRCode from canvas.
+ * @returns {Promise<string>}
+ */
+const getQrcode = async () => {
+  const result = await codeReader?.decodeFromCanvas(canvas.value as HTMLCanvasElement)
```

Comment:
Remove `await`, it does not return a Promise.

```suggestion
  const result = codeReader?.decodeFromCanvas(canvas.value as HTMLCanvasElement)
```


If you want to tell TS that this value is required at this stage, use **Non-null assertion operator** instead, but with caution. I don't recommend using this pattern a lot.

```suggestion
  const result = await codeReader?.decodeFromCanvas(canvas.value!)
```